src/sycl/CondFormats/SiPixelFedCablingMapGPUWrapper.h

```diff
@@ -0,0 +1,51 @@
+#ifndef RecoLocalTracker_SiPixelClusterizer_SiPixelFedCablingMapGPUWrapper_h
+#define RecoLocalTracker_SiPixelClusterizer_SiPixelFedCablingMapGPUWrapper_h
+
+#include "SYCLCore/ESProduct.h"
+#include "SYCLCore/device_unique_ptr.h"
+#include "CondFormats/SiPixelFedCablingMapGPU.h"
+
+#include <CL/sycl.hpp>
+
+#include <set>
+
+class SiPixelFedCablingMapGPUWrapper {
+public:
+  explicit SiPixelFedCablingMapGPUWrapper(SiPixelFedCablingMapGPU const &cablingMap,
+                                          std::vector<unsigned char> modToUnp);
+  ~SiPixelFedCablingMapGPUWrapper();
+
+  bool hasQuality() const { return hasQuality_; }
+
+  // returns pointer to GPU memory
+  const SiPixelFedCablingMapGPU *getGPUProductAsync(sycl::queue stream) const;
+
+  // returns pointer to GPU memory
+  const unsigned char *getModToUnpAllAsync(sycl::queue stream) const;
+
+private:
+  std::vector<unsigned char> modToUnpDefault;
+  bool hasQuality_;
+
+  SiPixelFedCablingMapGPU *cablingMapHost_ = nullptr;  // pointer to struct in CPU
+
+  struct GPUData {
+    GPUData() = default;
+    ~GPUData() {}
+
+    std::unique_ptr<SiPixelFedCablingMapGPU, cms::sycltools::device::impl::DeviceDeleter>
+        cablingMapDevice;  // pointer to struct in GPU
```

Comment:
Would
```suggestion
    cms::sycltools::device::unique_ptr<SiPixelFedCablingMapGPU> cablingMapDevice;  // pointer to struct in GPU
```
work here? (I haven't gotten yet to the definition of `cms::sycltools::device::unique_ptr`)

Reply:
yes, I'll change it. In `cms::sycltools::device`:
```cpp
template <typename T>
      using unique_ptr = std::unique_ptr<T, impl::DeviceDeleter>;
```